lsp-mode.el

```diff
@@ -6490,28 +6490,41 @@ should return the command to start the LS server."
   "Create LSP stdio connection named name.
 LOCAL-COMMAND is either list of strings, string or function which
 returns the command to execute."
+  ;; 2.5.0-pre (as built from native-comp branch before M Albinus released tramp-2.5)
+  ;; worked fine
+  (defvar tramp-version)
+  (defvar tramp-connection-properties)
+  (when (version< "2.5.0-pre" tramp-version)
+    (lsp-warn
+     "Your tramp version - %s - might fail to work with remote LSP. Update to tramp-2.5 for tested reliability improvements"
+     tramp-version))
+  ;; Force a direct asynchronous process.
+  (add-to-list 'tramp-connection-properties
+               (list (regexp-quote (file-remote-p default-directory))
+                     "direct-async-process" t))
   (list :connect (lambda (filter sentinel name environment-fn)
-                   (let* ((final-command (lsp-resolve-final-function local-command))
-                          ;; wrap with stty to disable converting \r to \n
+                   (let* ((final-command (lsp-resolve-final-function
+                                          local-command))
+                          (_stderr (or (when generate-error-file-fn
```

Comment:
this should be appended to the command. Otherwise, the stderr will end up on stdout. (Unless this is fixed in the latest tramp).

Reply:
what do you mean by `this` here?

Reply:
_stderr

Reply:
ping? The fix is just to concat `"2>" stderr` just like we are doing now to avoid stderr polluting the stdout(this was an issue with old tramp).

Reply:
According to the comment from fjl, this PR doesn't really fix the problem for him. 

I will change the method and then try to wire-up some integration tests with tramp - though I am not sure if that's possible on github actions

Reply:
The fix works for me with pyright, whereas upgrading tramp alone did not.  I do see it requires key based (non-prompting) access, which might be an inconvenience for some.